ynput/ayon-core
# Loader tool: Filter by statuses

## Changelog Description
Added combobox for filtering by statuses in loader tool.

## Additional info
Added multiselection combobox for statuses that can be used to filter versions in loader tool. Selection in combobox does show stauts icons in the combobox field. Combobox view also has 3 action items: Select all, Deselect all and Swap which swaps selection. These might be removed if it's too much.

We might add similar combobox for product types, but there would have to be added better implementation of paint for the combobox field as icons are not enough.

### Screenshots
![image](https://github.com/ynput/ayon-core/assets/43494761/2476cbdb-9474-439c-b2c6-a6fa01a82704)

![image](https://github.com/ynput/ayon-core/assets/43494761/f22d50f4-8c6c-42dd-af65-0c38619bcada)

## Testing notes:
1. Use loader from tray or use loader tool in host. It would be maybe better to test multiple hosts with Qt Nuke, Maya, Houdini etc. so we validate it looks and works there.


3. Select folder with products.
4. Change statuses filter > versions or whole products should start to disappear.

EDITED:
I changed behavior of the combobox, when nothing is selected it behaves as everything is selected. Everything is deselected by default, so user is actually checking statuses he wants to see instead of deselecting them.
Maybe with this new behavior we don't need the helper actions to de/select all and swap selection.
